### PR TITLE
Mamba doc update

### PR DESCRIPTION
**Pull request by facero**
**Description**

This PR is to adress the issue #3710. It adds a short description of mamba in the `getting started` and  `dev` docs as an alternative to conda.
While mamba can also be used as a full replacement for conda (e.g. `mamba activate gammapy-0.19`), I prefer to stick to use mamba just for installation to reduce confusion in users mind ;-)

I also updated the dev doc (intro.rst) to mention using `conda activate` by default instead of `source activate`.
The switch in conda happened in 2017. I imagine most of users have conda > 4.4 today.

### CODE REVIEW

**Review by registerrier — COMMENTED** [review]
Thanks @facero ! 
This will need a rebase now that PR #3722 has been merged.

File: docs/development/intro.rst [hunk 1]
```diff
@@ -135,9 +135,11 @@ commands to set up an environment for Gammapy development:
     git clone https://github.com/[your-github-username]/gammapy.git
     cd gammapy
     conda env create -f environment-dev.yml
-    source activate gammapy-dev
-    # for conda versions >=4.4.0 you may have to execute
-    #'conda activate gammapy-dev' instead
+    #To speed up the environment solving you can use mamba as an alternative
```
Comment (maxnoe) [inline docs/development/intro.rst:138]: Maybe link to the mamba docs here?

**Review by Bultako — CHANGES_REQUESTED** [review]
@facero 

The file `getting-started/quickstart.rst` has been renamed into `getting-started/index.rst` in the [new version of the docs](https://docs.gammapy.org/dev/getting-started/index.html).
So you can safely remove the `getting-started/quickstart.rst` file and modify the `getting-started/index.rst` file instead.

File: docs/development/intro.rst (hunk 1 above)
Comment (facero) [reply docs/development/intro.rst]: Done

**Comment by Bultako** [comment]
@facero 
please, remove the `getting-started/quickstart.rst` file from the PR, this file is not needed because it has been renamed :)

**Review by Bultako — APPROVED** [review]

**Comment by Bultako** [comment]
Thanks @facero !

**Pull request merged**